enhancements/insights/conditional-data-gathering.md

```diff
@@ -0,0 +1,199 @@
+---
+title: conditional-data-gathering
+authors:
+  - "@Sergey1011010"
+reviewers:
+  - "@inecas"
+  - "@tremes"
+  - "@smarterclayton"
+approvers:
+  - "@smarterclayton"
+creation-date: 2021-07-15
+last-updated: 2021-07-15
+status: implementable
+---
+
+# Conditional Data Gathering
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The conditional gatherer for Insights Operator collects data according to the defined rules*. 
+Each rule contains one or more conditions such as "alert A is firing" 
+and one or more gatherers with parameters such as "collect N lines of logs from containers in namespace N".
+Current version has these rules defined in the code and the proposal is to load them from an external source
+to make collection of new conditional data faster. It's NOT about running brand new code, 
+but just calling the existing gatherers with different validated parameters, so the operator can't 
+collect anything we don't expect. 
+
+\* note that rule here and later has nothing to do with rules used to analyze data in archives written 
+by CCX presentation team
+
+## Motivation
+
+Collecting data on some condition such as an alert is quite common pattern 
+during the root-cause analysis and the ability to define the rules 
+what to collect on which condition externally will help to do that faster.
+
+### Goals
+
+- Implement validation of the rules for conditional gatherer
+- Create a service providing these rules
+- Fetch them from there and apply to the conditional gatherer
```

Comment:
I would probably mention the Git repository here as the source of the gathering rules defintions. I aslo think we should be explicit so instead of saying:
```
Fetch them from there ....
```
we should say something like `Fetch the rules definitions from the service API` or something like that.

Reply:
fixed